backend/layers/persistence/persistence.py

```diff
@@ -0,0 +1,153 @@
+from datetime import datetime
+from typing import List
+from backend.corpora.common.entities.dataset import Dataset
+from backend.layers.common.entities import CollectionMetadata, CollectionVersion, DatasetArtifact, DatasetMetadata, DatasetStatus, DatasetVersion
+
+
+class DatabaseProviderInterface:
+        
+    def initialize_canonical_collection(self, owner: str, collection_metadata: CollectionMetadata) -> CollectionVersion:
+        """
+        Initializes a canonical collection, generating a canonical collection_id and a new version_id.
+        Returns the newly created CollectionVersion
+        """
+        pass
+
+
+    def get_collection_version(self, version_id: str) -> CollectionVersion:
+        """
+        Retrieves a specific collection version by id
+        """
+        pass
+
+    def get_all_versions_for_collection(self, collection_id: str) -> List[CollectionVersion]:
+        """
+        Retrieves all versions for a specific collections, without filtering
+        """
+        pass
+
+    def get_all_collections(self) -> Iterable[CollectionVersion]:
+        """
+        Retrieves all collections.
+        # TODO: add a filtering object.
+        """
+        pass
+
+    def delete_canonical_collection(self, collection_id: str) -> None:
+        """
+        Deletes (tombstones) a canonical collection.
+        """
+        pass
+
+    def save_collection_metadata(self, version_id: str, collection_metadata: CollectionMetadata) -> None:
+        """
+        Saves collection metadata for a collection version
+        """
+        pass
+
+    def save_collection_publisher_metadata(self, version_id: str, publisher_metadata: dict) -> None:
+        """
+        Saves publisher metadata for a collection version
+        """
+        pass
+
+    def add_collection_version(self, collection_id: str) -> CollectionVersion:
```

Comment:
we don't expect to pass an initial CollectionMetadata?

Reply:
The idea is to get it from the existing version. That said, we can add it as a parameter and let the business layer take care of that.

Reply:
Yeah, I don't see doing a copy-on-create as being useful. The business layer can decide what new data constitutes the new version (taking it from the previous version).